src/helpers.rs

```diff
@@ -6,24 +6,37 @@
  * copied, modified, or distributed except according to those terms.
  */
 
+use std::fmt::Write;
 use std::u16;
 use std::u32;
-use std::fmt::Write;
 
 /// the only visible character we escape
 pub(crate) const BSLASH: u8 = b'\\';
 pub(crate) const BSLASH_U16: u16 = BSLASH as u16;
 
 /// create `u8` from two bytes of hex
-pub(crate) fn from_hex2(hex2: &[u8]) -> u8 {
```

Comment:
can we use `u32::from_str_radix` instead of all these `from_hex` methods?

https://doc.rust-lang.org/std/primitive.u32.html#method.from_str_radix